src/connectors/clear-refinements/connectClearRefinements.js

```diff
@@ -89,70 +91,62 @@ export default function connectClearRefinements(renderFn, unmountFn = noop) {
 
-      init({ helper, instantSearchInstance, createURL }) {
-        const attributesToClear = getAttributesToClear({
-          helper,
-          includedAttributes,
-          excludedAttributes,
-          transformItems,
-        });
-        const hasRefinements = attributesToClear.length > 0;
-
-        this._refine = () => {
-          helper
-            .setState(
-              clearRefinements({
-                helper,
-                attributesToClear: getAttributesToClear({
-                  helper,
-                  includedAttributes,
-                  excludedAttributes,
-                  transformItems,
-                }),
-              })
-            )
-            .search();
-        };
-
-        this._createURL = () =>
-          createURL(
-            clearRefinements({
-              helper,
-              attributesToClear: getAttributesToClear({
-                helper,
-                includedAttributes,
-                excludedAttributes,
-                transformItems,
-              }),
-            })
-          );
-
+      init({ instantSearchInstance }) {
         renderFn(
           {
-            hasRefinements,
-            refine: this._refine,
-            createURL: this._createURL,
+            hasRefinements: false,
+            refine: noop,
+            createURL: noop,
             instantSearchInstance,
             widgetParams,
           },
           true
         );
       },
 
-      render({ helper, instantSearchInstance }) {
-        const attributesToClear = getAttributesToClear({
-          helper,
-          includedAttributes,
-          excludedAttributes,
-          transformItems,
-        });
-        const hasRefinements = attributesToClear.length > 0;
+      render({ scopedResults, createURL, instantSearchInstance }) {
+        const attributesToClear = scopedResults.reduce(
+          (results, scopedResult) => {
+            return results.concat(
+              getAttributesToClear({
+                scopedResult,
+                includedAttributes,
+                excludedAttributes,
+                transformItems,
+              })
+            );
+          },
+          []
+        );
 
         renderFn(
           {
-            hasRefinements,
-            refine: this._refine,
-            createURL: this._createURL,
+            hasRefinements: attributesToClear.some(
+              attributeToClear => attributeToClear.items.length > 0
+            ),
+            refine: () => {
+              attributesToClear.forEach(({ helper: indexHelper, items }) => {
+                indexHelper
+                  .setState(
+                    clearRefinements({
+                      helper: indexHelper,
+                      attributesToClear: items,
+                    })
+                  )
+                  .search();
+              });
+            },
+            createURL: () =>
+              createURL(
+                mergeSearchParameters(
```

Comment:
From what I've understood the implementation won't behave like expected. With the usage of `merge` we apply the state of the children on the current index. Only its URL of will be computed but this is not what we want. We want to compute the URL for the current index + siblings + children. With the current implementation of `createURL` I don't think it's possible to implement the expected behaviour. 

The idea would be to provide a list of "patch" to the `createURL` function. Those "patches" describe which indices have to be updated with their next `SearchParameters`. We're then able to compute the `uiState` of each of the index. By default the `createURL` would be called with one patch for the current index, when we have a widget that alter multiple indices we can provide multiple patches. This is the raw idea we can see how it would fit with the current architecture.

Reply:
This will be addressed in another PR because some prior work needs to be done to support `createURL` in a multi-index context.